transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/TransformationPipelineBuilder.scala

```diff
@@ -148,23 +148,39 @@ object TransformationPipelineBuilder extends PipelineBuilder[Args] {
       s"${args.outputPrefix}/experiment_activity"
     )
 
+    // Biosample transformation needs Libraries, Experiments, and BiosampleTypes
+    val biosampleInputs = readRawEntities(EncodeEntity.Biosample)
+    val biosampleTypeInputs = readRawEntities(EncodeEntity.BiosampleType)
+
+    val typesById = biosampleTypeInputs
+      .withName("Key biosample types by ID")
+      .keyBy(_.read[String]("@id"))
+
+    val biosamplesWithTypes = biosampleInputs
+      .withName("Key biosamples by type")
+      .keyBy(_.read[String]("biosample_ontology"))
+      .join(typesById)
```

Comment:
`join` is an inner join, so any biosamples w/o an ontology will be silently dropped. That _should_ be fine since we expect every sample to have a linked type, but I could see an argument for using an outer join here and having the transformation log if a `None` slips through

Reply:
I have convinced myself that silently dropping data is bad, so this should be changed